### PR TITLE
WEBDAM-15 -- Create upload form

### PR DESCRIPTION
**Pull request by anairamzap-mobomo**
Hey, I have: 
- created the form in admin/content/media/webdam
- added getFolder() method to be able to retrieve folder name (and I will use it later to get the assets for WEBDAM-18 too)
- Added "new" method in tests

Let me know if its ok,

m

### CODE REVIEW

**Comment by cweagans** [comment]
This needs unit tests, and we should be uploading the file to Webdam itself, not to public://media_webdam. Sorry for short feedback - getting on a plane.

**Comment by anairamzap-mobomo** [comment]
> This needs unit tests

Yep, still finding a bit tricky to create unit tests from scratch, but will provide one for the new form as soon as possible.

> and we should be uploading the file to Webdam itself, not to public://media_webdam

Oh, I thought we were going to upload to Webdam in a later dev stage when we extend the guzzle client... will try to figure it out and let you know if I get stuck.

**Comment by anairamzap-mobomo** [comment]
Added the test for the new form.
Getting problems testing the submitForm, so still working on it. The other methods are there and passing.
Will ping you later for the guzzle client extension task.

**Comment by cweagans** [comment]
I'll do a full review tomorrow, but a couple quick notes:

* Conflicts with src/Webdam.php
* In the test, no self:: calls. Use $this.
* The form depends on the folders_filter config key that no longer exists. You'll have to use the API to build a folder tree.
* Don't limit the file size in the form - let Drupal handle that.
* Don't save files to public:// -- use `file_directory_temp()`

**Comment by anairamzap-mobomo** [comment]
Hey Cameron, I've checked and the conflict is related to the new methods we needed in Webdam class (getFolder() and uploadAsset()) and the getAsset() you added in last commit. So I've merged and solved the conflicts now, and pushing this now.

File: src/Form/WebdamUpload.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      '#type' => 'fieldset',
+      '#title' => $this->t('Upload media'),
+    ];
+    $form['upload_media']['managed_file'] = [
+      '#type' => 'managed_file',
+      '#title' => $this->t('Media Upload'),
+      '#description' => $this->t('Select a file to Upload. Max upload size: 1MB'),
+      '#upload_location' => 'temporary://',
+      '#upload_validators' => [
+        'file_validate_extensions' => ['gif png jpg jpeg mp3 mp4 mkv'],
```
Comment (cweagans) [inline src/Form/WebdamUpload.php]: We probably don't need these.

File: src/Form/WebdamUpload.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      ],
+      '#multiple' => FALSE,
+      '#required' => TRUE,
+    ];
+    $form['upload_media']['webdam_folder'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Webdam folder'),
+      '#description' => $this->t('Please select a Webdam folder to store your file'),
+      '#options' => $this->webdam->getFlattenedFolderList(),
```
Comment (cweagans) [inline src/Form/WebdamUpload.php]: Can you add a @TODO here to change this from the flattened folder list to something a little nicer?

File: config/install/media_webdam.webdamupload.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+media_webdam:
```
Comment (cweagans) [inline config/install/media_webdam.webdamupload.yml]: Can you remove this file entirely?

File: media_webdam.routing.yml [hunk 4]
```diff
@@ -5,3 +5,11 @@ media_webdam.config:
...
   requirements:
     _permission: 'administer site configuration'
+
+media_webdam.webdam_upload:
+  path: '/admin/content/media/webdam'
+  defaults:
+    _form: '\Drupal\media_webdam\Form\WebdamUpload'
+    _title: 'Webdam Upload'
+  requirements:
+    _permission: 'administer site configuration'
```
Comment (cweagans) [inline media_webdam.routing.yml]: We'll need to change this permission later, but it's okay for now.

File: src/Form/WebdamUpload.php (hunk 2 above)
Comment (anairamzap-mobomo) [reply src/Form/WebdamUpload.php]: Sure. Do you mean the method name, right?

File: config/install/media_webdam.webdamupload.yml (hunk 3 above)
Comment (anairamzap-mobomo) [reply config/install/media_webdam.webdamupload.yml]: oops! can I add config/ dir to the .gitignore?

File: media_webdam.routing.yml (hunk 4 above)
Comment (anairamzap-mobomo) [reply media_webdam.routing.yml]: ok. Will add a #TODO comment as a reminder.

File: src/Form/WebdamUpload.php (hunk 2 above)
Comment (cweagans) [reply src/Form/WebdamUpload.php]: Yeah. We don't have an alternative right now, but when we do, we should use it.

File: config/install/media_webdam.webdamupload.yml (hunk 3 above)
Comment (cweagans) [reply config/install/media_webdam.webdamupload.yml]: No, just delete the file. We will eventually have other config that we need to add here.

**Comment by cweagans** [comment]
:+1:

**Pull request merged**